src/renderer/helpers/colorTools.ts

```diff
@@ -0,0 +1,49 @@
+// From https://stackoverflow.com/questions/5560248/programmatically-lighten-or-darken-a-hex-color-or-rgb-and-blend-colors
+
+/**
+ * Lightens (percentage > 0) or darkens (percentage < 0) the given hex color.
+ *
+ * The color has to be in the form `#RRGGBB`.
+ */
+export const shadeColor = (color: string, percent: number): `#${string}` => {
+    let R = parseInt(color.substring(1, 3), 16);
+    let G = parseInt(color.substring(3, 5), 16);
+    let B = parseInt(color.substring(5, 7), 16);
+
+    R = Math.min(Math.round(R * (1 + percent / 100)), 255);
+    G = Math.min(Math.round(G * (1 + percent / 100)), 255);
+    B = Math.min(Math.round(B * (1 + percent / 100)), 255);
+
+    const RR = R.toString(16).padStart(2, '0');
+    const GG = G.toString(16).padStart(2, '0');
+    const BB = B.toString(16).padStart(2, '0');
+
+    return `#${RR}${GG}${BB}`;
+};
+
+// Below from https://codepen.io/njmcode/pen/NWdYBy
+
+// Converts a #ffffff hex string into an [r,g,b] array
+const h2r = (hex: string): number[] => {
+    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return [parseInt(result![1], 16), parseInt(result![2], 16), parseInt(result![3], 16)];
```

Comment:
Assuming valid inputs is not a good strategy when it comes to parsing. Please throw an error when the hex is invalid/`result` is null.